### DISCUSSION TITLE
ValueError: [E024] Got invalid whitespace entity spans after using spacy debug data

### DISCUSSION THREAD

**Discussion by jacky0218** [explosion/spaCy#10523] [Help: Other Questions] [answered]
Hi, 
I got some error when training my Spacy NER model(in Japanese).
the error shows that:
ValueError: [E024] Could not find an optimal move to supervise the parser. Usually, this means that the model can't be updated in a way that's valid and satisfies the correct annotations specified in the GoldParse. For example, are all labels added to the model? If you're training a named entity recognizer, also make sure that none of your annotated entity spans have leading or trailing whitespace or punctuation. You can also use the `debug data` command to validate your JSON-formatted training data. For details, run:
python -m spacy debug data --help

After using 
!python -m spacy debug data /content/model-best/config.cfg

I got the 3 invalid whitespace entity spans in Named Entity Recognition like this:

ℹ 8 label(s)
0 missing value(s) (tokens with '-' label)
✘ 3 invalid whitespace entity spans
⚠ Low number of examples for label 'e_1' (17)
⚠ Low number of examples for label 'e_7' (49)
⚠ Low number of examples for label 'e_9' (21)
⚠ Low number of examples for label 'e_4' (12)
⚠ Low number of examples for label 'e_3' (33)
⚠ Low number of examples for label 'e_8' (39)
✔ Examples without occurrences available for all labels
✔ No entities crossing sentence boundaries
Entity spans consisting of or starting/ending with whitespace characters are
considered invalid.

Is there any solution to deal with it?
Thx

**Comment by adrianeboyd** [comment, marked as answer]
Usually fixing the invalid whitespace spans will resolve this problem. No entity spans should start or end with whitespace, or consist entirely of whitespace. It's just a hard-coded limitation of the `ner` component.

**Reply by jacky0218** [reply]
Is there any solution that can fix the invalid whitespace spans?

**Reply by adrianeboyd** [reply]
There's not a built-in solution to fix it for you automatically, but it shouldn't be hard to check your data. Here's an example: https://github.com/explosion/spaCy/discussions/9029#discussioncomment-1225508

As Paul mentions, it would be nicer if there was better support for this.
